zschema/compounds.py

```diff
@@ -192,8 +192,11 @@ def validate(self, value):
             raise DataValidationException("record is not a dict", str(value))
         for subkey, subvalue in value.items():
             if subkey not in self.definition:
-                raise DataValidationException("%s is not a valid subkey of root",
-                                              subkey)
+                port = Port(subkey)
```

Comment:
blergh. This really is a pretty non-generic, censys-specific change in an otherwise generic library.

I'm also not quite figuring out what problem lead to this? Can you elaborate?